pkg/gadgets/trace/open/tracer/tracer_test.go

```diff
@@ -0,0 +1,210 @@
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				config := &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, 0),
+				}
+				return createTracer(t, config, eventCallback)
+			},
+			GenerateEvent: generateEvent,
+			ValidateEvent: utilstest.ExpectNoEvent[types.Event, int],
+		},
+		"filtering_one_event_should_be_captured": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				config := &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+				return createTracer(t, config, eventCallback)
+			},
+			GenerateEvent: generateEvent,
+			ValidateEvent: utilstest.ExpectOneEvent(func(info *utilstest.Info, ret *int) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					MountNsID: info.MountNsID,
+					Pid:       uint32(info.Pid),
+					UID:       uint32(info.UID),
+					Comm:      info.Comm,
+					Fd:        *ret,
+					Ret:       *ret,
+					Err:       0,
+					Path:      "/dev/null",
+				}
+			}),
+		},
+		"event_has_correct_UID": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				config := &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+				return createTracer(t, config, eventCallback)
+			},
+			// 1435 is just a random unprivileged UID,
+			// there's nothing special about this number.
+			RunnerConfig:  &utilstest.Config{UID: 1435},
+			GenerateEvent: generateEvent,
+			ValidateEvent: utilstest.ExpectOneEvent(func(info *utilstest.Info, ret *int) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					MountNsID: info.MountNsID,
+					Pid:       uint32(info.Pid),
+					UID:       uint32(info.UID),
+					Comm:      info.Comm,
+					Fd:        *ret,
+					Ret:       *ret,
+					Err:       0,
+					Path:      "/dev/null",
+				}
+			}),
+		},
+		"event_has_correct_error": {
+			GetTracer: func(info *utilstest.Info, eventCallback func(event types.Event)) utilstest.Tracer {
+				config := &tracer.Config{
+					MountnsMap: utilstest.CreateMntNsFilterMap(t, info.MountNsID),
+				}
+				return createTracer(t, config, eventCallback)
+			},
+			GenerateEvent: func() (int, error) {
+				_, err := unix.Open("non-existing-file", 0, 0)
+				if err == nil {
+					return 0, fmt.Errorf("error was expected")
+				}
+
+				return -1, nil
+			},
+			ValidateEvent: utilstest.ExpectOneEvent(func(info *utilstest.Info, _ *int) *types.Event {
+				return &types.Event{
+					Event: eventtypes.Event{
+						Type: eventtypes.NORMAL,
+					},
+					MountNsID: info.MountNsID,
+					Pid:       uint32(info.Pid),
+					UID:       uint32(info.UID),
+					Comm:      info.Comm,
+					Fd:        0,
+					Ret:       -int(unix.ENOENT),
+					Err:       int(unix.ENOENT),
```

Comment:
It seems 2 properties are tested here, ideally that would be 2 different subtests.